### PR TITLE
Use whitelist model for plugin class loading

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

**Comment by martint** [comment]
looks good, but check build failures.

File: presto-main/src/main/java/com/facebook/presto/server/PluginClassLoader.java [hunk 1]
```diff
@@ -15,52 +15,44 @@
...
-    private final List<String> parentFirstClasses;
-    private final List<String> hiddenResources;
-    private final List<String> parentFirstResources;
-
-    public PluginClassLoader(List<URL> urls,
-            ClassLoader parent,
-            Iterable<String> hiddenClasses,
-            Iterable<String> parentFirstClasses)
+    private final ClassLoader spiClassLoader;
+    private final List<String> spiClasses;
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/server/PluginClassLoader.java]: This should be named `spiPackages` or `spiPrefixes`

**Comment by dain** [comment]
Product tests are throwing:
```
Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.apache.hadoop.hdfs.DFSConfigKeys

**Comment by electrum** [comment]
Yes, this is due to a dependency on `javax.servlet`, fixed upstream in https://issues.apache.org/jira/browse/HDFS-8101

We need to update to Hadoop 2.7.3 https://github.com/prestodb/presto-hadoop-apache2/pull/17

**Comment by dain** [comment]
Assuming travis passes, looks good

**Pull request closed without merging**